onboarding/source/amb_light_service.c

```diff
@@ -17,37 +21,126 @@
 
 /* USER CODE BEGIN */
 // Define light service queue config here
+#define QUEUE_LENGTH 10
+#define ITEM_SIZE sizeof(uint64_t)
 
 /* USER CODE END */
 
 /* USER CODE BEGIN */
 // Declare any global variables here
+static TaskHandle_t lightServiceTaskHandle = NULL;
+static StaticTask_t lightServiceTaskBuffer;
+static StackType_t lightServiceTaskStack[LIGHT_SERVICE_STACK_SIZE];
 
+static QueueHandle_t queueHandle = NULL;
+static StaticQueue_t queueBuffer;
+uint8_t ucQueueStorageArea[QUEUE_LENGTH * ITEM_SIZE];
 /* USER CODE END */
 
 /**
  * @brief Task that waits for MEASURE_LIGHT event and then prints the ambient light value to the serial port.
  * @param pvParameters Task parameters
  */
-static void lightServiceTask(void * pvParameters);
+static void lightServiceTask(void *pvParameters);
 
-obc_error_code_t initLightService(void) {
+obc_error_code_t initLightService(void)
+{
     /* USER CODE BEGIN */
     // Create the task and queue here. Return error code if task/queue was not created successfully.
+    if (lightServiceTaskHandle == NULL)
+    {
+        lightServiceTaskHandle = xTaskCreateStatic(lightServiceTask,
+                                                   LIGHT_SERVICE_NAME,
+                                                   LIGHT_SERVICE_STACK_SIZE,
+                                                   NULL,
+                                                   LIGHT_SERVICE_PRIORITY,
+                                                   lightServiceTaskStack,
+                                                   &lightServiceTaskBuffer);
+    }
+
+    if (lightServiceTaskHandle == NULL)
+    {
+        return OBC_ERR_CODE_TASK_CREATION_FAILED;
+    }
+
+    if (queueHandle == NULL)
+    {
+        queueHandle = xQueueCreateStatic(QUEUE_LENGTH,
+                                         ITEM_SIZE,
+                                         ucQueueStorageArea,
+                                         &queueBuffer);
+    }
 
+    if (queueHandle == NULL)
+    {
+        return OBC_ERR_CODE_QUEUE_CREATION_FAILED;
+    }
+
+    return OBC_ERR_CODE_SUCCESS;
     /* USER CODE END */
 }
 
-static void lightServiceTask(void * pvParameters) {
+uint32_t getLightSensorData(void)
+{
+    adcData_t adc_data;
+
+    /** - Start Group1 ADC Conversion
+     *     Select Channel 6 - Light Sensor for Conversion
+     */
+    adcInit();
+
+    /** - Wait for ADC Group1 conversion to complete */
+
+    adcStartConversion(adcREG1, adcGROUP1);
+    /** - Read the conversion result
+     *     The data contains the Light sensor data
+     */
+    while (!adcIsConversionComplete(adcREG1, adcGROUP1))
+        ;
+
+    /** - Transmit the Conversion data to PC using SCI
```

Comment:
It'd make more sense to move this comment to line 122 or just remove it altogether.